eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/NodeRelay.scala

```diff
@@ -178,14 +181,15 @@ class NodeRelay private(nodeParams: NodeParams,
    * @param nextPayload relay instructions (should be identical across HTLCs in this set).
    * @param nextPacket  trampoline onion to relay to the next trampoline node.
    * @param handler     actor handling the aggregation of the incoming HTLC set.
+   * @param triggered   async payment trigger received
    */
-  private def receiving(htlcs: Queue[UpdateAddHtlc], nextPayload: IntermediatePayload.NodeRelay.Standard, nextPacket: OnionRoutingPacket, handler: ActorRef): Behavior[Command] =
+  private def receiving(htlcs: Queue[UpdateAddHtlc], nextPayload: IntermediatePayload.NodeRelay.Standard, nextPacket: OnionRoutingPacket, handler: ActorRef, triggered: Boolean = false): Behavior[Command] =
```

Comment:
I don't think we should handle the trigger here, this won't always work.
In this phase, we are receiving incoming htlcs before relaying.
Imagine we have the following sequence of events:

- we receive the first incoming htlc, but not the second one (so we're not ready to relay)
- the recipient comes online, so we receive the trigger
- the recipient goes back offline
- we receive the second incoming htlc

We will try to relay instantly whereas the recipient has gone offline.

I think the only change to this behavior should be to go to `waitForTrigger` instead of `doSend` when the payload says it's an async payment.

Reply:
That makes sense. Instead of dealing with a trigger that comes before the `waitForTrigger` state in `NodeRelay` it makes more sense that when `waitForTrigger` registers the trigger callback it will immediately get a trigger if the destination node is already connected, or later when the destination node connects.

Reply:
This is fixed in [96acc3a](https://github.com/ACINQ/eclair/pull/2435/commits/96acc3aaff05cac3c81714c865b6745e5b70df3a)